diktat-rules/src/test/kotlin/org/cqfn/diktat/util/FixTestBase.kt

```diff
@@ -58,6 +59,64 @@ open class FixTestBase(protected val resourceFilePath: String,
         )
     }
 
+    private fun getSaveForCurrentOs() = when {
+        System.getProperty("os.name").startsWith("Linux", ignoreCase = true) -> "save-$SAVE_VERSION-linuxX64.kexe"
+        System.getProperty("os.name").startsWith("Mac", ignoreCase = true) -> "save-$SAVE_VERSION-macosX64.kexe"
+        System.getProperty("os.name").startsWith("Windows", ignoreCase = true) -> "save-$SAVE_VERSION-mingwX64.exe"
+        else -> ""
+    }
+
+    private fun createProcessBuilderWithCmd(expectedPath: String, testPath: String): ProcessBuilder {
+        val filesDir = "src/test/resources/test/smoke"
+        val savePath = "$filesDir/${getSaveForCurrentOs()}"
+
+        val systemName = System.getProperty("os.name")
+        val result = when {
+            systemName.startsWith("Linux", ignoreCase = true) || systemName.startsWith("Mac", ignoreCase = true) ->
+                ProcessBuilder("sh", "-c", "pwd ; chmod 777 $savePath ; ./$savePath src/test/resources/test/smoke/src/main/kotlin $testPath --log all")
+            else -> ProcessBuilder(savePath, "src/test/resources/test/smoke/src/main/kotlin", expectedPath, testPath)
+        }
+        return result
+    }
+
+    /**
+     * @param expectedPath path to file with expected result, relative to [resourceFilePath]
+     * @param testPath path to file with code that will be transformed by formatter, relative to [resourceFilePath]
+     * @param configFilePath path of diktat-analysis file
+     */
+    @Suppress("TOO_LONG_FUNCTION")
+    protected fun saveSmokeTest(
```

Comment:
Why is it located in `FixTestBase`? I would expect to find this method and the other one for regular smoke test in corresponding smoke test class implementations.

Reply:
Fixed